### PR TITLE
Add mark and unmark command

### PR DESCRIPTION
**Pull request by Angelyxx**
Allows users to mark and unmark contacts of interest.

Marked users (after executing `mark`) will have a solid star (★) indicated in their person card, ☆ otherwise.

Passes all test cases.

### CODE REVIEW

**Review by geoffong11 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -119,6 +118,20 @@ public void deletePerson(Person target) {
...
+        updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+    }
+
+    @Override
+    public void unMarkPerson(Person person) {
+        addressBook.unMarkPerson(person);
+        updateFilteredPersonList(PREDICATE_SHOW_NO_PERSONS);
+        updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+    }
+
```
Comment (xavierpok) [inline src/main/java/seedu/address/model/ModelManager.java:134]: I'm only slightly concerned about the `updateFilteredPersonList` call, but I suppose `addPerson` does the same thing, so I suppose we can talk about `addPerson` and these lines together in a meeting!

Lines 131, 124, the calls to `updateFilteredPersonList(PREDICATE_SHOW_NO_PERSONS);` maybe redundant given `addPerson` only updates to show all?

File: src/main/java/seedu/address/model/person/Mark.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents the mark status in the address book.
+ * Guarantees: mutable.
+ */
+public class Mark {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Mark Status can only be true or false!";
```
Comment (xavierpok) [inline src/main/java/seedu/address/model/person/Mark.java:11]: More minor nits to imply variable type. Probably not needed, but if you want!
```suggestion
    public static final String MESSAGE_CONSTRAINTS =
            "Mark Status can only be a boolean of true or false!";
```

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ public class Person {
     private final Company company;
     private final Job job;
     private final Set<Tag> tags = new HashSet<>();
-
+    private final Mark markStatus = new Mark(false);
```
Comment (xavierpok) [inline src/main/java/seedu/address/model/person/Person.java:28]: Mayb could move this into constructor?

Alternatively, maybe could edit constructor to take `Mark` as an argument?

Given that the default state is initialised to `Mark` being false, though, I don't think this is a big deal!

**Review by xavierpok — APPROVED** [review]
LGTM overall, just some minor nits regarding a few strings and code structure. Overall, looks good!

**Pull request merged**